tests/functional/base_test.go

```diff
@@ -52,6 +52,33 @@ func DefaultDataPlaneSpec() dataplanev1.OpenStackDataPlaneSpec {
 	}
 }
 
+func DataPlaneWithNodeSpec() dataplanev1.OpenStackDataPlaneSpec {
+	return dataplanev1.OpenStackDataPlaneSpec{
+		DeployStrategy: dataplanev1.DeployStrategySection{
+			Deploy: false,
+		},
+		Roles: map[string]dataplanev1.OpenStackDataPlaneRoleSpec{
+			"edpm-compute": {
+				Services: []string{"configure-network", "validate-network", "install-os", "configure-os", "run-os"},
+				NodeTemplate: dataplanev1.NodeSection{
+					AnsibleSSHPrivateKeySecret: "dataplane-ansible-ssh-private-key-secret",
+				},
+			},
+		},
+		Nodes: map[string]dataplanev1.OpenStackDataPlaneNodeSpec{
+			"edpm-compute-0": {
+				Node: dataplanev1.NodeSection{
+					NetworkConfig: dataplanev1.NetworkConfigSection{
+						Template: "{'test-nic-override-value}",
```

Comment:
I can't really think of a scenario where a customer _wont_ be using a custom nic-config. So I think it's fine if we just leave this in this function rather than creating a dedicated spec for testing nic-config overrides.

Reply:
If users are always going to use inline jinja templates (using static config probably would not possible with IPAM and other things) we could get rid of `NicConfigTemplateFile` field completely and create/mount the file every time.  The other option could be customers use a custom ansibleee image for the network_config service which has nic config templates already included in it.